File: LeetCode/stack/JZ09TwoStacksToImpQueue.py
class CQueue(object):

    def __init__(self):
        self.stack1 = []
        self.stack2 = []

    def appendTail(self, value: int) -> None:
        """
        :type value: int
        :rtype: None
        """
        while self.stack1:
            self.stack2.append(self.stack1.pop())
        self.stack1.append(value)
        while self.stack2:
            self.stack1.append(self.stack2.pop())

    def deleteHead(self)-> int:
        """
        :rtype: int
        """
        if not self.stack1: return  -1
        return self.stack2.pop()

File: LeetCode/stack/JZ09TwoStacksToImpQueue.java
package LeetCode.stack;

import java.util.LinkedList;

public class JZ09TwoStacksToImpQueue {
    class CQueue {
        LinkedList<Integer> stack1, stack2;

        public CQueue() {
            stack1 = new LinkedList<>();
            stack2 = new LinkedList<>();
        }

        public void appendTail(int value) {
            stack1.addLast(value);
        }

        public int deleteHead() {
            if (!stack2.isEmpty()) return stack2.removeLast();
            if (stack1.isEmpty()) return -1;
            while (!stack1.isEmpty()) {
                stack2.addLast(stack1.removeLast());
            }

            return stack2.removeLast();
        }
    }
}
